backend/src/main/scala/sbt/internal/inc/bloop/BloopZincCompiler.scala

```diff
@@ -173,15 +172,18 @@ object BloopZincCompiler {
       setup: MiniSetup,
       previousSetup: Option[MiniSetup],
       ignoredScalacOptions: Array[String],
-      sources: Set[File],
+      sources: Set[VirtualFile],
       previousAnalysis: CompileAnalysis,
       manager: ClassFileManager,
       logger: ObservedLogger[_]
   ): CompileAnalysis = {
     // Copied from `Incremental` to pass in the class file manager we want
-    def prune(invalidatedSrcs: Set[File], previous0: CompileAnalysis, classfileManager: ClassFileManager): Analysis = {
+    def prune(invalidatedSrcs: Set[VirtualFile], previous0: CompileAnalysis, classfileManager: ClassFileManager): Analysis = {
       val previous = previous0 match { case a: Analysis => a }
-      classfileManager.delete(invalidatedSrcs.flatMap(previous.relations.products).toArray)
+      val toDelete = invalidatedSrcs.flatMap(previous.relations.products).toArray.collect {
+        case vf: VirtualFile => vf
```

Comment:
Can we get anything else then VirtualFile here (like a entry in the jar)? If so, shouldn't we somehow invalidate

Reply:
We should not really have anything else, classFileManager only works with VirtualFile, so I don't imagine anything else is normally handled.